lib/spack/spack/cmd/uninstall.py

```diff
@@ -317,9 +310,15 @@ def uninstall_specs(args, specs):
     if not args.yes_to_all:
         confirm_removal(anything_to_do)
 
-    # just force-remove things in the remove list
-    for spec in remove_list:
-        _remove_from_env(spec, env)
+    if env:
+        # Remove all the specs that are supposed to be uninstalled or just
+        # removed.
+        with env.write_transaction():
+            for spec in remove_list:
+                _remove_from_env(spec, env)
+            for spec in uninstall_list:
+                _remove_from_env(spec, env)
```

Comment:
I wanted to make sure you were OK with the updated semantics of merging these two removal (this will result in differing behavior when there is a failure according to https://github.com/spack/spack/pull/14676#issue-368741563). I assume this means you agree so I agree this can (should) be aggregated into one loop.